source/reflect.h2

```diff
@@ -1501,13 +1662,38 @@ union: (inout t : meta::type_declaration)
 
 //-----------------------------------------------------------------------
 //
-//  print - output a pretty-printed visualization of t
+//  print - output a pretty-printed visualization of t or f
 //
 print: (t: meta::type_declaration) =
 {
     std::cout << t.print() << "\n";
 }
 
+print: (f: meta::function_declaration) =
+{
+    std::cout << f.print() << "\n";
+}
+
+
+//-----------------------------------------------------------------------
+//
+//  meta - mark f as a metafunction and generate registration code for it
+//
+meta: (inout f: meta::function_declaration) =
+{
+    // TODO(DyXel): Check that the function:
+    //  * has ::cpp2::meta::function_declaration as only parameter, OR,
+    //  * has ::cpp2::meta::type_declaration as only paramater, OR,
+    //  * function_declaration or type_declaration (implicitly deduced to be the above), AND,
+    //  * either in or inout paramater passing style, AND,
+    //  * to be publicly accessible, AND,
+    //  * to not be on header file, AND,
```

Comment:
They could be in a header file. If this header is just included in one translation unit it would be valid code.

Reply:
Yeah, tbh I am not sure if we should allow it or not, in any case we can detect multiple registration of the same function, which is a clear ODR violation, but it can be entirely avoided if we just prevent users from writing stuff in headers in the first place.

Reply:
Ok. Maybe have extra file extension for defining metafunctions.

Registering a function twice should not be a problem if the function pointer is the same. Having the same name with a different function pointer should be a problem.